### ISSUE TITLE
future warnings

### ISSUE THREAD

**Issue by solegalli** [feature-engine/feature_engine#693]
List of future pandas warnings

```
c:\users\sole\documents\repositories\feature_engine\feature_engine\creation\math_features.py:212: FutureWarning: The provided callable <function sum at 0x0000016E1CA6D090> is currently using Series.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass 'sum' instead.
```

**Comment by glevv** [comment]
Connected to this
https://github.com/pandas-dev/pandas/issues/53425

I think we can easily suppress this warning or not so easily bump pandas version.

We should also add information to this transformer, that same numpy and pandas functions could lead to different results (this is why they raise this warning). For example, `np.std` has `ddof=0` while `Series.std()` has `ddof=1`

**Comment by olikra** [comment]
I sorted out the > 3000 warning messages of a complete pytest-run. Most of them ('SkipTestWarning', 'UserWarning', 'ConvergenceWarning', 'RuntimeWarning') can be irgnored by now to get a better view whats relevant.

In the feature-engine code itself we have to take care for:

- FutureWarning 24 issues
- DeprecationWarnings 2 issues
- PytestReturnNotNoneWarning 3 issues

For details look into this file:
[pytest_warnings.csv](https://github.com/user-attachments/files/15989006/pytest_warnings.csv)

In #774 i already worked on math_features - so it makes sense for me taking care of it. The rest needs to get sorted out as well cause we saw last week what happens ignoring / not seeing this warnings(#772).

**Comment by solegalli** [comment]
Thank you @olikra That would be great!

**Comment by solegalli** [comment]
Hey @olikra 

Thank you so much for your work on the repo. I've got a quick question:

Did you take care of all the warnings on #777 ?

**Comment by solegalli** [comment]
closed in #781
